Echo/ApplicationInputHandler.cpp

```diff
@@ -38,99 +37,151 @@ bool ApplicationInputHandler::optionsParse( const int argc, char *argv[] )
 
     opterr = 0;
 
-    while ( ( optionArgumentIndex = getopt( argc, argv, "o:f:hH") ) != -1 )
+    int messageFilterCounter = 0;
+
+    while ( ( optionArgumentIndex = getopt( argc, argv, "t:o:f:hH") ) != -1 )
     {
         optionIndex = getFlagIndex( (const char) optionArgumentIndex );
 
-        if ( optionIndex == -1 )
+        if( optionIndex == -1 )
         {
             cout << "\n\nUsage: invalid options. Usage guide follows." <<endl;
 
             _getOptHelpFlag = true;
         }
-
-        else if ( optionIndex != -1 )
+        else if( optionIndex != -1 )
         {
-            switch ( optionArgumentIndex )
+            switch( optionArgumentIndex )
             {
+                case 'f':
+
+                    if( ( *argv[ optind - 1 ] )
+                         && ( *argv[ optind - 1 ] == '-' ) )
+                    {
+                        cout <<"\n\nInvalid usage for option -f filter single message type:\n"
+                             <<"-f should be followed by a PolySync message type,"
+                             << " not by another -option.\n"
+                             <<"A usage guide follows." << endl;
+
+                        _getOptHelpFlag = true;
+                    }
+                    else
+                    {
+                        if( messageFilterCounter == 0 )
+                        {
+                            _messageName = optarg;
+
+                            _filteredForSingleMessageFlag = true;
+
+                            ++messageFilterCounter;
+                        }
+                        else
+                        {
+                            _filteredForMultipleMessagesFlag = true;
+
+                            _multipleFilteredMessageNames.emplace_back( optarg );
+                        }
```

Comment:
Possibly it could, yes. getopt() passes through the argument list vector several times, and using this logic allows for registering of a single listener versus registering for multiple listeners, with same -f option.